core/extension/gdextension.cpp

```diff
@@ -337,6 +337,65 @@ void GDExtension::_register_extension_class(GDExtensionClassLibraryPtr p_library
 
 	ClassDB::register_extension_class(&extension->gdextension);
 }
+
+void GDExtension::_register_extension_class2(GDExtensionClassLibraryPtr p_library, GDExtensionConstStringNamePtr p_class_name, GDExtensionConstStringNamePtr p_parent_class_name, const GDExtensionClassCreationInfo2 *p_extension_funcs) {
```

Comment:
Rather than maintaining both `_register_extension_class()` and `_register_extension_class2()` with nearly identical code, I think `_register_extension_class()` should be made to build the new `GDExtensionClassCreationInfo2` struct and call `_register_extension_class2()`.